### PR TITLE
upgrade to jetty 9.4

### PR DESCRIPTION
**Pull request by olamy**
Signed-off-by: olivier lamy <olamy@apache.org>

### CODE REVIEW

**Comment by olamy** [comment]
@jglick @oleg-nenashev please review guys thanks!

File: pom.xml [hunk 1]
```diff
@@ -47,7 +47,10 @@ THE SOFTWARE.
 
   <properties>
     <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
-    <jetty.version>9.2.15.v20160210</jetty.version>
+    <jetty.version>9.4.5.v20170502</jetty.version>
+    <maven-surefire-plugin.version>2.20</maven-surefire-plugin.version>
+    <maven-compiler-plugin.version>3.6.1</maven-compiler-plugin.version>
```
Comment (oleg-nenashev) [inline pom.xml]: Why would you need the explicit versions here?

Comment (jglick) [inline pom.xml]: Also curious about this. The parent POM defines these versions IIUC.

**Comment by jglick** [comment]
Not sure why CI build is broken; opened #64 to modernize it.

File: pom.xml (hunk 1 above)
Comment (olamy) [inline pom.xml]: well it's really to use more up2date versions :-)

File: src/main/java/org/jvnet/hudson/test/JenkinsRule.java [hunk 2]
```diff
@@ -685,9 +686,11 @@ public Thread newThread(Runnable r) {
     protected LoginService configureUserRealm() {
         HashLoginService realm = new HashLoginService();
         realm.setName("default");   // this is the magic realm name to make it effective on everywhere
-        realm.update("alice", new Password("alice"), new String[]{"user","female"});
-        realm.update("bob", new Password("bob"), new String[]{"user","male"});
-        realm.update("charlie", new Password("charlie"), new String[]{"user","male"});
+        UserStore userStore = new UserStore();
```
Comment (jglick) [inline src/main/java/org/jvnet/hudson/test/JenkinsRule.java]: Forced by Jetty changes as detailed in https://github.com/eclipse/jetty.project/pull/1488#issuecomment-332201285. This makes it awkward to switch Jetty versions in a plugin back to 9.2.x, which @vivek says is currently necessary for Wiremock compatibility: you must use

```java
@Rule
public JenkinsRule r = new JenkinsRule() {
    @Override
    protected LoginService configureUserRealm() {
        HashLoginService realm = new HashLoginService();
        realm.setName("default");
        return realm;
    }
};
```

We cannot simply delete the `UserStore` part from the default implementation; it works for most tests, but not those calling `WebClient.login` while using `LegacySecurityRealm` (perhaps indirectly via `@PresetData(DataSet.NO_ANONYMOUS_READACCESS)`), of which there are a few in core and some plugins. (For example, `JnlpAccessWithSecuredHudsonTest`, using `HudsonTestCase`.)

**Pull request merged**